<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1089</width>
    <height>646</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget#centralwidget{background-color: rgb(127,220,165)}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QRadioButton" name="useASMdll">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>510</y>
      <width>191</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Convert using ASM library</string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">dllGroup</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="useCPPdll">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>470</y>
      <width>191</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>Convert using C++ library</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">dllGroup</string>
    </attribute>
   </widget>
   <widget class="QLabel" name="imageBefore">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>110</y>
      <width>441</width>
      <height>291</height>
     </rect>
    </property>
    <property name="acceptDrops">
     <bool>true</bool>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="scaledContents">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="imageAfter">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>110</y>
      <width>441</width>
      <height>291</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="labelBefore">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>80</y>
      <width>211</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:11pt; color:#5555ff;&quot;&gt;Image before conversion&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelAfter">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>80</y>
      <width>221</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:11pt; color:#5555ff;&quot;&gt;Image after conversion&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="loadImageButton">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>200</y>
      <width>91</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Load image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="convertImageButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>240</y>
      <width>91</width>
      <height>24</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>Convert</string>
    </property>
   </widget>
   <widget class="QLabel" name="title">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>441</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:14pt; font-weight:700; font-style:italic; color:#aa55ff;&quot;&gt;Project name: Converting image to greyscale&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="messageForUser">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>270</y>
      <width>111</width>
      <height>141</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Waiting for action...&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="labelButton">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>110</y>
      <width>441</width>
      <height>291</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: transparent;</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="nameLabel">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>410</y>
      <width>251</width>
      <height>31</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="dimensionsLabel">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>440</y>
      <width>251</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QCheckBox" name="scaleImagesButton">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>440</y>
      <width>181</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Scale image to fit frame</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="timeLabel">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>470</y>
      <width>251</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>460</y>
      <width>341</width>
      <height>91</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>10</y>
       <width>261</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string>How many threads to convert image:</string>
     </property>
    </widget>
    <widget class="QLabel" name="threadsDisplay">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>0</y>
       <width>31</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QSlider" name="threadsSlider">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>40</y>
       <width>201</width>
       <height>31</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>64</number>
     </property>
     <property name="pageStep">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="showGreyHistogramButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>570</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Grey histogram</string>
    </property>
   </widget>
   <widget class="QPushButton" name="showColorHistogramButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>570</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Color histogram</string>
    </property>
   </widget>
   <zorder>frame</zorder>
   <zorder>imageBefore</zorder>
   <zorder>useASMdll</zorder>
   <zorder>useCPPdll</zorder>
   <zorder>imageAfter</zorder>
   <zorder>labelBefore</zorder>
   <zorder>labelAfter</zorder>
   <zorder>loadImageButton</zorder>
   <zorder>convertImageButton</zorder>
   <zorder>title</zorder>
   <zorder>messageForUser</zorder>
   <zorder>labelButton</zorder>
   <zorder>nameLabel</zorder>
   <zorder>dimensionsLabel</zorder>
   <zorder>scaleImagesButton</zorder>
   <zorder>timeLabel</zorder>
   <zorder>showGreyHistogramButton</zorder>
   <zorder>showColorHistogramButton</zorder>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>loadImageButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>421</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>370</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>convertImageButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>convertImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>535</x>
     <y>251</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>labelButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>250</x>
     <y>255</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>threadsSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>threadsSlider()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>540</x>
     <y>536</y>
    </hint>
    <hint type="destinationlabel">
     <x>542</x>
     <y>304</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showGreyHistogramButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>CreateGreyChart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>535</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>636</x>
     <y>482</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showColorHistogramButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>CreateColorChart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>645</x>
     <y>581</y>
    </hint>
    <hint type="destinationlabel">
     <x>544</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>loadImage()</slot>
  <slot>convertImage()</slot>
  <slot>threadsSlider()</slot>
  <slot>CreateColorChart()</slot>
  <slot>CreateGreyChart()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="dllGroup"/>
 </buttongroups>
</ui>
